src/pages/login/login.test.tsx

```diff
@@ -39,6 +40,29 @@ test('triggers login on form submit', () => {
   });
 });
 
+test('triggers login error on form submit', () => {
+  const mockAxiosResponse: AxiosResponse = {
+    data: {
+      non_field_errors: 'Unable to log in with provided credentials.',
+    },
+    status: 1,
+    statusText: '',
+    headers: {},
+    config: {},
+  };
+  const mockAxiosError: AxiosError = {
+    config: {},
+    message: '',
+    name: '',
+    response: mockAxiosResponse,
+  };
+  const view = shallow(<Login {...props} error={mockAxiosError} />);
+  const preventDefault = jest.fn();
+  view.find('form').simulate('submit', { preventDefault });
+  expect(props.login).toBeCalled();
+  expect(view.find('.pf-c-alert')).not.toBeNull();
```

Comment:
You can actually do this: 
```
import {  Alert } from '@patternfly/react-core';
...
expect(view.find(Alert).exists()).toBe(true);
```
While I am sure it is working currently the`.not.toBeNull()` is actually not a valid check here.  enzyme returns a collection no matter the result of finding something.  If it isn't there was nothing found it's length is just 0.

Reply:
Additionally it might be nice to assert the text is correct. A shortcut for this would be
```
expect(view.find(Alert)).toMatchSnapshot();
```
That will verify the correct alert type, the correct testid and the text.

Reply:
Thanks, I actually had something similar to that, but didn't work properly because I wasn't using `.exists`